spacy_llm/tasks/ner.py

```diff
@@ -168,9 +170,10 @@ def initialize(
             for initialization.
         nlp (Language): Language instance.
         labels (List[str]): Optional list of labels.
+        gather_examples (bool): Whether to gather examples for the task. False by default.
         """
 
-        examples = get_examples()
+        examples = list(get_examples())
```

Comment:
Instead of putting all the `examples` in a list, we can rewrite the following lines to only loop through all `examples` once, and for each one add to the `label_set` if we need to deduce labels, and create the prompt_example and add it to the internal list if `gather_examples` (`gather_prompt_examples`?) is `True`.

Reply:
Agreed.